src/full/Agda/TypeChecking/Serialise.hs

```diff
@@ -197,7 +197,7 @@ decode s = do
      else do
 
       st <- St (ar nL) (ar ltL) (ar stL) (ar bL) (ar iL) (ar dL)
-              <$> liftIO H.new
+              <$> liftIO (H.initialize 0)
```

Comment:
I think it would be better to have a compatibility shim ([like we had for HashMap](https://github.com/agda/agda/commit/2014ba8b60f85451973818b8289f5e8844fbd2f1#diff-e5de80dd54070b15824e39272f54d8cf112e5b217d95a43c6adc167bc2b224f5L1)) so that changes to switch to a different hashtable provider are confined to that module.